src/transformStateWithClones.js

```diff
@@ -65,7 +65,60 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const obj = { ...state };
+  const arr = [];
+
+  for (let i = 0; i < transforms.length; i++) {
+    const { operation, properties } = transforms[i];
+
+    if (operation === 'addProperties') {
+      for (const key in properties) {
+        obj[key] = properties[key];
+      }
+    } else if (operation === 'removeProperties') {
+      for (let item = 0; item < properties.length; item++) {
+        const current = properties[item];
+
+        if (obj[current]) {
+          delete obj[current];
+        }
+      }
+    } else if (operation === 'clear') {
+      for (const member in obj) {
+        delete obj[member];
+      }
+    }
+
+    const objTest = { ...obj };
+
+    arr.push(objTest);
```

Comment:
I think that creating one more copy of 'obj' is quite unnecessary,
`arr.push({ ...obj });` would be enough

Reply:
Being short is not my strength))

Thank you!

Reply:
Yes, Artem saying po delu. Pofixi pliz